src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1916,10 +2163,15 @@ open_common(int (*real_open)(const char *pathname, int oflags, ...), const char
 		return real_open(pathname, oflags);
 	else
 		return real_open(pathname, oflags, mode);
+
 out_error:
 	FREE(parent_dir);
 	errno = rc;
 	return (-1);
+
+out_compatible:
```

Comment:
it sounds like error path can leave different leaked stuff and need better error handling